platforms/core-configuration/kotlin-dsl-tooling-builders/src/crossVersionTest/groovy/org/gradle/kotlin/dsl/tooling/builders/r88/KotlinSettingsScriptModelCrossVersionSpec.groovy

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.tooling.builders.r88
+
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.kotlin.dsl.tooling.builders.AbstractKotlinScriptModelCrossVersionTest
+
+@TargetGradleVersion(">=8.8")
+class KotlinSettingsScriptModelCrossVersionSpec extends AbstractKotlinScriptModelCrossVersionTest {
+
+    def "settings script has type-safe accessors on the classpath"() {
+        given:
+        withDefaultSettingsIn("included")
+        withBuildScriptIn("included", """
+            plugins { `kotlin-dsl` }
+            $repositoriesBlock
+        """)
+        file("included/src/main/kotlin/SettingsExtension.kt") << """
+            import org.gradle.api.provider.*
+            interface SettingsExtension {
+                val myProperty: Property<Int>
+            }
+        """
+        file("included/src/main/kotlin/settings-plugin.settings.gradle.kts") << """
+            import org.gradle.kotlin.dsl.create
+            extensions.create<SettingsExtension>("mySettingsExtension")
+        """
+        withSettings("""
+            pluginManagement {
+                includeBuild("included")
+            }
+            plugins {
+                id("settings-plugin")
+            }
+        """)
+
+        when:
+        def accessorsClassPath = accessorsClassPathFor(settingsFileKts)
+
+        then:
+        !accessorsClassPath.isEmpty()
```

Comment:
@eskatos Are we good with such an assertion?

Reply:
It feels weird somehow to assert this way.
But I can't think of a better way that is not overkill so I'm tempted to say this is acceptable.